### ISSUE TITLE
Add option to disable double clicking to add a search engine that uses HTTP POST request

### ISSUE THREAD

**Issue by kapitainsky** [odebroqueville/contextSearch#197]
An image is worth a thousand words so short video will be even better:

https://github.com/user-attachments/assets/c3b32fcd-f0f7-4ed5-9ffd-b3405c3be419

not even sure what exact words describe this contextSearch behaviour. Double clicking at empty text box triggers "something".
I have not found anything in settings to disable it. Double click at empty form and voila - magic contextSearch window opens. This trick is amusing initially but becomes bothersome when happening again and again.

For reference I used this site: https://eclecticlight.co/2024/11/07/which-m4-chip-and-model/

But what I can see so far it happens on any site with form where Firefox suggests some previously entered values. It does not happen when nothing is suggested as either Firefox has nothing to offer or this action is suppressed by site itself (as this site search at the top of this page)

**Comment by odebroqueville** [comment]
@kapitainsky It's not a bug. As laid out in the README note (see last bullet point below), it's a feature!! 

<img width="879" alt="CleanShot 2024-11-10 at 16 03 04@2x" src="https://github.com/user-attachments/assets/a8858826-0137-4f53-a65d-4dbc7bcbb4b4">

You're the first users out of 5'000 users and 10 years of existence to complain about it.

Would you please explain in what circumstances you need to double-click in a form.

**Comment by kapitainsky** [comment]
> You're the first user out of 5'000 users and 10 years of existence to complain about it.

hahaha. Sorry for that. Must be my way of interacting with text fields.

> Would you please explain in what circumstances you need to double-click in a form.

I do often without even thinking to show drop down list of suggestions (previously entered values) so I can choose correct one. This is what double click does..

For sure it is not something I can not live with. So if I am the only one complaining and everybody thinks that it is normal behaviour then let it be. 

Still would be nice IMO to be able to turn it off but I do understand it is only my wish with very low priority.

**Comment by odebroqueville** [comment]
Ok, I'll keep this on my radar, but like you said: it's not a high priority. Please bare in mind that you won't be able to add search engines using HTTP POST requests if you decided to disable this feature.

**Comment by serrufs** [comment]
I agree, this is also convenient for me, I just didn’t notice it before, I started noticing it for about a month, in general, I would like to turn it off - I don’t understand why this is even necessary?

**Comment by kapitainsky** [comment]
> I agree, this bothers me too, I just didn’t notice it before, I started noticing it for about a month,

Yes this is weird as I could swear I never saw it before and only started seeing after v5.3 was released... about a month ago. 
Really annoying that it works now (maybe it was happily broken earlier):)

**Comment by BobsACoder** [comment]
Please, odebroqueville, add an option in the settings to disable this behavior!

It can be extremely annoying. It occurs even when trying to hit the up- or down-arrows on a number field.

**Comment by kapitainsky** [comment]
Good to hear that it is not only me who finds it annoying:) And indeed the best option would be if it is configurable - I do understand that others might use it.

**Comment by odebroqueville** [comment]
Sorry for the delay. I've been very busy preparing v6 based on manifest v3 which will future proof the extension. Still trying to get it working for Chromium-based browsers as well. In the mean time, please give release 5.3.5 a try and let me know if it addresses your needs or if you still see glitches.

<img width="505" alt="CleanShot 2024-12-16 at 19 50 33@2x" src="https://github.com/user-attachments/assets/f1369640-c2ab-4f48-90d4-6218b559baa0" />

P.S. I'm not sure yet when v6 will be released (possibly not before the new year), but I'd strongly recommend that you start making an export of your search engines for backup purposes in case anything goes wrong! Merry X-mas to all.

**Comment by kapitainsky** [comment]
This is  great news. v5.3.5 works and indeed allows double click to be turned off. 

Thank you very much for adding this option.

And Merry Christmas to all too.

**Comment by kapitainsky** [comment]
As it is done I am closing this issue.

**Comment by odebroqueville** [comment]
@h0bbyte Thanks for spotting this! What must have happened is that the latest changes in 5.3.5 were left out in 6+. I'll have a look into this and provide a fix asap. Sorry for the trouble.

**Comment by h0bbyte** [comment]
@odebroqueville , thank you for your response and for working on this excellent extension.

**Comment by h0bbyte** [comment]
@odebroqueville, yes, the double-click on/off setting for the dialog box works in this release. Great work, thank you.
